WorkspaceServer/Kernel/RenderingEngine.cs

```diff
@@ -12,21 +12,30 @@ public class RenderingEngine : IRenderingEngine
     {
         private readonly IRenderer _defaultRenderer;
         private readonly IRendering _nullRendering;
-        private readonly Dictionary<Type,IRenderer> _rendererRegistry = new Dictionary<Type, IRenderer>();
+        private readonly Dictionary<Type, IRenderer> _rendererRegistry = new Dictionary<Type, IRenderer>();
 
         public RenderingEngine(IRenderer defaultRenderer, IRendering nullRendering)
         {
             _defaultRenderer = defaultRenderer;
             _nullRendering = nullRendering;
         }
+
         public IRendering Render(object source)
         {
-            if (source == null)
+            try
             {
-                return _nullRendering;
+                if (source == null)
+                {
+                    return _nullRendering;
+                }
+
+                var renderer = FindRenderer(source.GetType());
+                return renderer.Render(source, this);
+            }
+            catch (Exception e)
```

Comment:
Is there a specific reason (e.g. expected exception type) to do this? `catch Exception` is something that should typically only ever happen at a much higher level in the code.

Reply:
this one is wrapping as well, and capturing the evaluation result